diff --git a/8.0/Dockerfile b/8.0/Dockerfile
index 9e4115a..aec2f69 100644
--- a/8.0/Dockerfile
+++ b/8.0/Dockerfile
@@ -14,6 +14,8 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
 FROM oraclelinux:7-slim
 
+#RUN groupadd -g 10000 -r mysql && useradd -r -u 10000 -g mysql mysql
+
 ARG MYSQL_SERVER_PACKAGE=mysql-community-server-minimal-8.0.19
 ARG MYSQL_SHELL_PACKAGE=mysql-shell-8.0.19
 
@@ -26,10 +28,13 @@ RUN yum install -y https://repo.mysql.com/mysql-community-minimal-release-el7.rp
       $MYSQL_SHELL_PACKAGE \
       libpwquality \
   && yum clean all \
-  && mkdir /docker-entrypoint-initdb.d
+  && mkdir /docker-entrypoint-initdb.d \
+  && mkdir -p /var/lib/mysql \
+  && chown -R mysql:mysql /var/lib/mysql
 
 VOLUME /var/lib/mysql
 
+
 COPY docker-entrypoint.sh /entrypoint.sh
 COPY healthcheck.sh /healthcheck.sh
 ENTRYPOINT ["/entrypoint.sh"]
diff --git a/8.0/docker-entrypoint.sh b/8.0/docker-entrypoint.sh
index fb6333b..9358b5a 100755
--- a/8.0/docker-entrypoint.sh
+++ b/8.0/docker-entrypoint.sh
@@ -46,12 +46,12 @@ if [ "$1" = 'mysqld' ]; then
 	DATADIR="$(_get_config 'datadir' "$@")"
 	SOCKET="$(_get_config 'socket' "$@")"
 
-	if [ -n "$MYSQL_LOG_CONSOLE" ] || [ -n "console" ]; then
+#	if [ -n "$MYSQL_LOG_CONSOLE" ] || [ -n "console" ]; then
 		# Don't touch bind-mounted config files
-		if ! cat /proc/1/mounts | grep "etc/my.cnf"; then
-			sed -i 's/^log-error=/#&/' /etc/my.cnf
-		fi
-	fi
+#		if ! cat /proc/1/mounts | grep "etc/my.cnf"; then
+#			sed -i 's/^log-error=/#&/' /etc/my.cnf
+#		fi
+#	fi
 
 	if [ ! -d "$DATADIR/mysql" ]; then
 		# If the password variable is a filename we use the contents of the file. We
@@ -69,8 +69,8 @@ if [ "$1" = 'mysqld' ]; then
 			MYSQL_RANDOM_ROOT_PASSWORD=true
 			MYSQL_ONETIME_PASSWORD=true
 		fi
-		mkdir -p "$DATADIR"
-		chown -R mysql:mysql "$DATADIR"
+#		mkdir -p "$DATADIR"
+#		chown -R mysql:mysql "$DATADIR"
 
 		echo '[Entrypoint] Initializing database'
 		"$@" --initialize-insecure
@@ -192,15 +192,15 @@ EOF
 	# Used by healthcheck to make sure it doesn't mistakenly report container
 	# healthy during startup
 	# Put the password into the temporary config file
-	touch /healthcheck.cnf
-	cat >"/healthcheck.cnf" <<EOF
+	touch /tmp/healthcheck.cnf
+	cat >"/tmp/healthcheck.cnf" <<EOF
 [client]
 user=healthchecker
 socket=${SOCKET}
 password=healthcheckpass
 EOF
-	touch /mysql-init-complete
-	chown -R mysql:mysql "$DATADIR"
+	touch /tmp/mysql-init-complete
+#	chown -R mysql:mysql "$DATADIR"
 	echo "[Entrypoint] Starting MySQL 8.0.19-1.1.15"
 fi
 
diff --git a/8.0/healthcheck.sh b/8.0/healthcheck.sh
index a3af378..55ba696 100755
--- a/8.0/healthcheck.sh
+++ b/8.0/healthcheck.sh
@@ -16,9 +16,9 @@
 
 # The mysql-init-complete file is touched by the entrypoint file before the
 # main server process is started
-if [ -f /mysql-init-complete ]; # The entrypoint script touches this file
+if [ -f /tmp/mysql-init-complete ]; # The entrypoint script touches this file
 then # Ping server to see if it is ready
-  mysqladmin --defaults-extra-file=/healthcheck.cnf ping
+  mysqladmin --defaults-extra-file=/tmp/healthcheck.cnf ping
 else # Initialization still in progress
   exit 1
 fi
